webhooks/pod_webhook.go

```diff
@@ -230,57 +221,116 @@ func (m *PodMutator) injectSidecar(
+		if source.Provider == "" {
+			source.Provider = flagSourceConfig.DefaultSyncProvider
+		}
+
+		var sourceCfg types.SourceConfig
+		var err error
+
+		switch {
+		case source.Provider.IsKubernetes():
+			sourceCfg, err = m.toKubernetesConfig(ctx, pod, source)
+			if err != nil {
+				return []types.SourceConfig{}, err
+			}
+		case source.Provider.IsFilepath():
+			sourceCfg, err = m.handleFilepathProvider(ctx, pod, sidecar, source)
+			if err != nil {
+				return []types.SourceConfig{}, err
+			}
+		case source.Provider.IsHttp():
+			sourceCfg = m.toHttpProviderConfig(source)
+		case source.Provider.IsGrpc():
+			sourceCfg = m.toGrpcProviderConfig(source)
+		default:
+			return []types.SourceConfig{}, fmt.Errorf("unrecognized sync provider in config: %s", source.Provider)
+		}
+
+		sourceCfgCollection = append(sourceCfgCollection, sourceCfg)
+
+	}
+
+	return sourceCfgCollection, nil
+}
+
+// toHttpProviderConfig generate types.SourceConfig for http provider
+func (m *PodMutator) toHttpProviderConfig(source v1alpha1.Source) types.SourceConfig {
+	return types.SourceConfig{
+		URI:         source.Source,
+		Provider:    string(v1alpha1.SyncProviderHttp),
+		BearerToken: source.HttpSyncBearerToken,
 	}
 }
 
-func (m *PodMutator) handleKubernetesProvider(ctx context.Context, pod *corev1.Pod, sidecar *corev1.Container, source v1alpha1.Source) error {
+// toGrpcProviderConfig generate types.SourceConfig for grpc provider
+func (m *PodMutator) toGrpcProviderConfig(source v1alpha1.Source) types.SourceConfig {
+	return types.SourceConfig{
+		URI:        source.Source,
+		Provider:   string(v1alpha1.SyncProviderGrpc),
+		CertPath:   source.CertPath,
+		ProviderID: source.ProviderID,
+		Selector:   source.Selector,
+	}
+}
+
+// toKubernetesConfig generate types.SourceConfig for K8s provider. Further, this updates underlying pod permissions &
+// annotations
+func (m *PodMutator) toKubernetesConfig(ctx context.Context,
+	pod *corev1.Pod, source v1alpha1.Source) (types.SourceConfig, error) {
+
 	ns, n := parseAnnotation(source.Source, pod.Namespace)
+
 	// ensure that the FeatureFlagConfiguration exists
 	ff := m.getFeatureFlag(ctx, ns, n)
 	if ff.Name == "" {
-		return fmt.Errorf("feature flag configuration %s/%s not found", ns, n)
+		return types.SourceConfig{}, fmt.Errorf("feature flag configuration %s/%s not found", ns, n)
 	}
+
 	// add permissions to pod
 	if err := m.enableClusterRoleBinding(ctx, pod); err != nil {
-		return err
+		return types.SourceConfig{}, err
 	}
+
 	// mark pod with annotation (required to backfill permissions if they are dropped)
 	pod.Annotations[fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, AllowKubernetesSyncAnnotation)] = "true"
-	// append args
-	sidecar.Args = append(
-		sidecar.Args,
-		"--uri",
-		fmt.Sprintf(
-			"core.openfeature.dev/%s/%s",
-			ns,
-			n,
-		),
-	)
+
+	// build K8s config
+	return types.SourceConfig{
+		URI:      fmt.Sprintf("core.openfeature.dev/%s/%s", ns, n),
```

Comment:
these prefixes can be removed, the first thing we do in flagd is remove them anyway

Reply:
Yes, the confusion here comes from flagd's way of handling configurations [1]. Right now it can work with prefix or without it :) And tests also validates this 


[1] - https://github.com/open-feature/flagd/blob/dc9cb5fa2005630fd68c7bc0a63a2b056430ff0d/core/pkg/runtime/from_config.go#L192